src/components/modals/index.ts

```diff
@@ -0,0 +1,10 @@
+import Modal from "./modal";
+
+export interface ModalProps {
+    isOpen: boolean;
+    title:string;
+    onToggle: React.Dispatch<React.SetStateAction<boolean>>;
+    fields: Array<string>;
```

Comment:
What does fields represent.
What if I want to have a modal that just contains  a single text element as a child?. Or what if I want to have a datepicker as a child element?.
You should introduce a children prop and render in out properly.